src/psyclone/domain/lfric/lfric_extract_driver_creator.py

```diff
@@ -0,0 +1,974 @@
+          kernel extraction does the same and stores the values under the name
+          ``f``, so the driver similarly simplifies the name back to the
+          original ``f``.
+          The KernCallArgList class will have enforced the appropriate
+          basic Fortran type declaration for each reference to a user defined
+          variable. For example, if a field ``f`` is used, the reference to
+          ``f_proxy%data`` will have a data type attribute of a 1D real array
+          (with the correct precision).
+
+    6. We create the read-in code for all variables in the input- and output-
+       lists. Mostly, no special handling of argument type is required (since
+       the generic interface will make sure to call the appropriate function).
+       But in case of user-defined types, we need to make sure to use the
+       name with '%' (unless it is a standard LFRic type like field which
+       as described above has already been simplified). Example are names like
+       ``f_proxy%local_stencil`` and ``f_proxy%ncell_3d``. They will be using
+       the name with ``%``, but a flattened variable name (``%`` replaced with
+       ``_``).
+
+       a. Input parameter are read in from the PSyData ReadKernelData module.
+          These function will allocate all array variables to the right size
+          based on the data from the input file.
+       b. For parameters that are read and written, two variables will be
+          declared: the input will be stored in the unmodified variable name,
+          and the output values in a variable with ``_post`` appended. For
+          example, a field ``f`` as input will be read into ``f`` using the
+          name ``f``, and output values will be read into ``f_post`` using
+          the name ``f_post``. The symbol table will make sure that the
+          ``_post`` name is unique.
+       c. Similar to b., output only parameters will be read into a variable
+          named with '_post' attached, e.g. output field ``f`` will be stored
+          in a variable ``f_post``. Then the array ``f`` is allocated based on
+          the shape of ``f_post`` and initialised to 0 (since it's an
+          output-only parameter the value doesn't really matter).
+
+    7. The extracted kernels are added to the program. Since in step 5 all
+       references have been replaced, there created code will use the correct
+       new variable names (which just have been read in). The output variables
+       with ``_post`` attached will not be used at all so far.
+    8. After the kernel calls are executed, each output variable is compared
+       with the value stored in the corresponding ``_post`` variable. For
+       example, a variable ``f`` which was modified in the kernel calls, will
+       then be compared with ``f_post``.
+
+    '''
+    def __init__(self, precision=None):
+        self._all_field_types = ["field_type", "integer_field_type",
```

Comment:
I've opened #2069 for this: I need to double check that replacing this (which would add operators to the list) does not introduce inconsistencies with the extraction code (since the names must be consistent between extraction and driver. E.g. at this stage both extraction and driver remove 'proxy' from LFRic fields so that the user sees the familiar names).